packages/e2e/test/wallet/SingleAddressWallet/delegation.test.ts

```diff
@@ -151,13 +151,6 @@ describe('SingleAddressWallet/delegation', () => {
     expect(tx1PendingState.balance.deposit).toEqual(stakeKeyDeposit);
 
     expect(tx1ConfirmedState.rewardAccount.delegatee?.nextNextEpoch!.id).toEqual(poolId);
-    // nothing changes for 2 epochs
-    expect(tx1ConfirmedState.rewardAccount.delegatee?.nextEpoch).toEqual(
-      initialState.rewardAccount?.delegatee?.nextEpoch
-    );
-    expect(tx1ConfirmedState.rewardAccount.delegatee?.currentEpoch).toEqual(
-      initialState.rewardAccount?.delegatee?.currentEpoch
-    );
```

Comment:
> will fail since tx1ConfirmedState currentEpoch and nextEpoch will be one greater than initialState currentEpoch and nextEpoch.

`delegatee` is typed as this:
https://github.com/input-output-hk/cardano-js-sdk/blob/2a4ccb0abead34481e817f807850d29e77d7340a/packages/wallet/src/services/types.ts#L75-L83

... so the assertion could be refactored to check if the epoch has already rolled over, and use the `currentEpoch` to compare with the `nextEpoch` for example.

> Also, this current assert checks that nothing changes for two epochs, but isn't this guaranteed by the rules of the network? (seems to be an assert on the network behavior)

The intent of these assertions is to check that our state reflects the domain rules, being that delegations are not active for two epochs.

Reply:
addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/424/commits/ef3462834c092294ee4f15176be6e2dc72b953f4